### DISCUSSION TITLE
Dropdown Trigger events

### DISCUSSION THREAD

**Discussion by akumari1812** [h2oai/wave#1390] [Q&A] [answered]
I am creating a dropdown using ui.from_card. So, I want whenever something is selected from the dropdown, it will show the data using ui.text in the same form card. I don't want to show in some another meta card. The dropdown and the data that I want to show after the trigger must be in the same card.

**Comment by azim-b** [comment, marked as answer]
if I understand your question correctly, you're asking how to render text from a `dropdown` trigger in the **same** form_card.

To do this, simply set the `content` of the text to `q.args.<dropdown_name>`. To handle for `None`, you could just say the content is `q.args.<dropdown_name>` or `''`.

---


Example: theme changer while displaying the current theme's name as text in the same form_card.

https://user-images.githubusercontent.com/64787868/165383120-3996e8c4-c1c3-44eb-876d-7980b8fd44cc.mov


Code:

```
q.page['example'] = ui.form_card('body', items=[
    ui.text_xl('H2O Wave'),
    ui.separator(),
    ui.label('Please select a theme'),
    ui.text(f'Current theme: {q.args.theme}' if q.args.theme else 'Current theme: monokai'),
    ui.dropdown(
        name='theme',
        label='Theme',
        value=q.args.theme,
        trigger=True,
        choices=[ui.choice(theme, theme) for theme in ['default', 'neon', 'h2o-dark', 'benext', 'ember', 'fuchasia']]
    )
])
```

**Reply by akumari1812** [reply]
So like this, I want to show a table on the same form card. and whenever I select an item from dropdown, the content of table should change on the basis of selected item.

**Reply by azim-b** [reply]
like I showed above, you could just set `trigger` to `True` for the dropdown, handle the dropdown args and do some work, update the table, and save the page.

```
# get the value of dropdown selection

# do some work

# create a ui.table()

# render dropdown and table 
```


---


Example: Select a team from the dropdown to see team members in a table dynamically



https://user-images.githubusercontent.com/64787868/165778741-789d5c3c-07cf-453e-bb1a-2fbd11a70a0d.mov



Code:

```
async def page(q:Q):
    # default dropdown selection
    team_selected = q.args.team or 'Engineering'

    # make a `ui.table()` for the selected team
    team_table = await get_team_table(team_selected)

    # render form card with team selection dropdown and team table
    q.page['main'] = ui.form_card('main', items=[
        # dropdown
        ui.inline(justify='center', items=[
            ui.dropdown(
                name='team',
                label='Team',
                trigger=True,
                value=team_selected,
                choices=[ui.choice(c, c) for c in ['Sales', 'Engineering', 'DevOps', 'Exec']],
                # optional UI enhancements
                required=True,
                width='500px',
            )
        ]),
        # table
        ui.separator(f'Displaying team {team_selected}'),
        team_table
    ])

    await q.page.save()




async def get_team_table(team_selected:str):
    '''Util function to create ui.table() for a team'''
    # get list of team members
    team_members = teams[team_selected]
    table = ui.table(
        name='team',
        # columns are [Name, Department]
        columns = [ui.table_column('name', 'Name'), ui.table_column('team', 'Team')],
        # each row is [Team Member, Department] for all team members of the selected team
        rows = [ui.table_row(team_member, [team_member, team_selected]) for team_member in team_members]
    )
    return table
```

**Comment by azim-b** [comment]
@akumari1812 was this helpful?

**Reply by akumari1812** [reply]
yes, this was helpful, Thank you so much.
